### PR TITLE
Update MS to use new Balance Proof

### PR DESCRIPTION
**Pull request by pcppcp**
- integrates contract changes
- adds a happy path e2e test
- removes obsolete contract tests

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,4 +31,4 @@ jobs:
         - flake8 monitoring_service
 
       script:
-        - coverage run --source monitoring_service/ --omit 'monitoring_service/test/*' -m py.test --travis-fold=always -vvvvvvvvs $TEST_TYPE
```
Comment (palango) [inline .travis.yml]: Do you still want to keep the `-x` after the discussion in RC?

File: monitoring_service/server.py [hunk 2]
```diff
@@ -93,45 +99,56 @@ def _run(self):
...
-        self.blockchain.add_confirmed_listener(
-            EVENT_TRANSFER_UPDATED,
-            lambda event, tx: self.on_transfer_updated(event, tx)
-        )
 
         # this loop will wait until spawned greenlets complete
         while self.stop_event.is_set() is False:
             tasks = gevent.wait(self.task_list, timeout=5, count=1)
             if len(tasks) == 0:
-                gevent.sleep(5)
```
Comment (palango) [inline monitoring_service/server.py]: Why did this change?

File: monitoring_service/server.py [hunk 3]
```diff
@@ -93,45 +99,56 @@ def _run(self):
...
+                gevent.sleep(1)
                 continue
             task = tasks[0]
+            log.info('%s completed (%s)' % (task, task.value))
```
Comment (palango) [inline monitoring_service/server.py]: nitpick: I remember reading somewhere that we should use the style: `log.info('%s completed (%s)', task, task.value)`

File: monitoring_service/tasks/on_channel_close.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            monitor_request.non_closing_signer,
+            balance_proof.balance_hash,
+            balance_proof.nonce,
+            balance_proof.additional_hash,
+            balance_proof.signature,
+            monitor_request.non_closing_signature,
+            monitor_request.reward_amount,
+            balance_proof.token_network_address,
+            monitor_request.reward_proof_signature
+        ).transact({'gas_limit': 350000}, private_key=private_key)
```
Comment (palango) [inline monitoring_service/tasks/on_channel_close.py]: where is this limit coming from?

File: monitoring_service/tasks/on_channel_close.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        receipt = web3.eth.getTransactionReceipt(tx_hash)
+        tx = web3.eth.getTransactionReceipt(tx_hash)
+        log.info(receipt)
```
Comment (palango) [inline monitoring_service/tasks/on_channel_close.py]: maybe it would be helpful to add more context to these log calls.

File: requirements.txt [hunk 6]
```diff
@@ -3,12 +3,11 @@ click
...
 gevent
 jsonschema
-eth-utils
-coincurve
-ethereum
-eth-tester
-web3>=4.0.0b1
+eth-utils>=1.0.1,<2.0
+coincurve>=7.1.0,<8.0
+ethereum>=2.3.1,<3.0
```
Comment (palango) [inline requirements.txt]: Just wondering: is this still needed?

Comment (pcppcp) [reply requirements.txt]: not sure, these dependencies were copypasted from `raiden-libs`

File: monitoring_service/tasks/on_channel_close.py (hunk 4 above)
Comment (pcppcp) [reply monitoring_service/tasks/on_channel_close.py]: guesswork :) `eth-tester` couldn't estimate gas properly for this

File: monitoring_service/server.py (hunk 3 above)
Comment (pcppcp) [reply monitoring_service/server.py]: this will come when I switch over to `structlog` here too

File: monitoring_service/server.py (hunk 2 above)
Comment (pcppcp) [reply monitoring_service/server.py]: shorter poll interval on test chain

File: .travis.yml (hunk 1 above)
Comment (pcppcp) [reply .travis.yml]: In order to have a shorter build push-CI breaks loop I'll leave it here for now. It can be removed once the code stabilizes.

**Pull request merged**